flag_text_test.go

```diff
@@ -0,0 +1,125 @@
+package cli
+
+import (
+	"errors"
+	"flag"
+	"io"
+	"log/slog"
+	"slices"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type badMarshaller struct{}
+
+func (badMarshaller) UnmarshalText(_ []byte) error {
+	return nil
+}
+
+func (badMarshaller) MarshalText() ([]byte, error) {
+	return nil, errors.New("bad")
+}
+
+func TestTextFlag(t *testing.T) {
+	tests := []struct {
+		name    string
+		flag    TextFlag
+		args    []string
+		want    string
+		wantErr bool
+	}{
+		{
+			name: "empty",
+			flag: TextFlag{
+				Name:  "log-level",
+				Value: &slog.LevelVar{},
```

Comment:
Note that Value is used to initialize the flag. The actual value of field should be in Destination

Reply:
See the flag_generic.go as an example.

Reply:
I've address this in the latest version of the PR.